packages/block-editor-tools/src/hooks/use-post-by-id/index.js

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+import usePost from '../use-post';
+/**
+ * Gets post data for a specific post given its ID. The post type is
+ * looked up from the search endpoint, cached, and then passed to usePost.
+ *
+ * @param int    postId   The ID for the post to return.
+ * @returns {object} An object containing a hasResolved property
+ *                   and the returned post object.
+ */
+const usePostById = (postId) => {
+  const [postTypeCache, setPostTypeCache] = useState({});
+
+  useEffect(() => {
+    if (!postId) {
+      return;
+    }
+    if (postTypeCache[postId]) {
+      return;
+    }
+    (async () => {
+      const path = addQueryArgs('/wp/v2/search', { include: postId });
```

Comment:
Here's where we could put in the logic to fetch from a custom function provided as an optional second parameter to the hook. We'd also need to keep track of whether that function fetched the post's data, because if it did, we'd want to return that instead of `usePost` below.